### DISCUSSION TITLE
Sorting/Filtering of followed projects and notifications

### DISCUSSION THREAD

**Discussion by MopsTMC** [modrinth/.github#28] [Feature requests]
### Is your feature request related to a problem? Please describe.

Navigating through followed projects and notifications is a bit tedious when looking for something of a specific type or modloader, especially now with the additions of resource packs and plugins.

### Describe the solution you'd like

1. Having the ability to sort either or both of those lists in Alphabetic order, Followed time, Recently Updated, Popularity etc. similar to that when searching for items
2. Filtering functionality based off of version and modloader
3. Tabs could be used for individual project types just like in the search page header for easier navigation when navigating trough Followed and Notifications

### Describe alternatives you've considered

Instead of tabs, filters similar to those when searching in the rewrite version of the website could be used as an alternative although I think that wouldn't be as practical.
Perhaps a combination of the two could lead somewhere better.

**Comment by RetroDestroyer** [comment]
Even something like the User Profile of creators where it's seperated into Mods, Modpacks, etc. Will be great to have for the followed tab.

**Comment by Boy0000** [comment]
Moving this here from: https://github.com/modrinth/knossos/issues/1034

### Is your suggested enhancement related to a problem? Please describe.

Right now the updates-list is horrible to look through.
Due to several projects releasing 20 versions per update (though this has been asked for in an  open issue already)
and that there is no split between different project types.
Suggest splitting these into tabs, or some form of filter atleast to make it alot easier to track stuff

![image](https://user-images.githubusercontent.com/62521371/224357786-43b5745f-3d55-4729-b173-e8f783144a43.png)


### Describe the solution you'd like

Either have project types split into tabs, or some filter like all the actual search pages use
![image](https://user-images.githubusercontent.com/62521371/224361287-c010a3cc-08da-4a52-b725-f66609fe683c.png)
![image](https://user-images.githubusercontent.com/62521371/224361335-0dcfae07-96d5-4a04-823e-2521a96a5814.png)

Im no UI genius but functionally this would just make so much more sense to me rather than bundling everything on one page
-------------------

Revisiting this as I went away for a few weeks. Came back to update my mods to find the Notification section unusable.
Hundreds of notifications, full of mods/plugins releasing for 1.12->1.19.x and all 3 mod loaders and so on
Creates scenarios where you get 40 notifications that you slowly have to remove one by one.

Not to single out the mod-authors here, as I believe the issue is solely with Modrinth, but this is an example of why the Notifications screen is more or less useless at this point.
![image](https://user-images.githubusercontent.com/62521371/232899388-dd3af7d2-6120-4e75-ad10-150f6837cdc4.png)

Additionally some projects have beta-builds which build per-commit. Adding to the above I would love to see another filter to remove beta-builds. Or optionally fully toggle them from being shown based on user-preference settings
![image](https://user-images.githubusercontent.com/62521371/232900420-8486d097-8c0f-4688-aa4e-7a5b4178c268.png)

I would like to add that the Discovery on Modrinth is so far above competition like CurseForge (which is actually not usable).
I probably come of as very pissy above sorry about that, but I believe that Notifications here are a big issue and would greatly help Modrinth.
Discovery's implementation is more or less exactly what i would like to see, ofc some changes are gonna need to be made

**Comment by Lafolie** [comment]
I would like to chime in and say that I think notifications should be filterable by modloader type. Personally, I am only interested in Fabric mods, so the notifications list seems cluttered with duplicate entries due to both Fabric/Forge/etc versions being published. Grouping multiple updates for a particular mod would also aid navigation.

Even without filtering, it would be great if there was a prefix or suffix along the lines of `[Fabric] Modname has been updated!`. 

Even better, perhaps this could integrate with a profile setting whereby users can select which modloaders they wish to see content for, thereby automatically filtering both the notifications and general search results. One issue I could foresee with this is that it may reduce visibility for say, Forge-only mods, when the user has selected only Fabric on their profile. A solution to this may be to have a clear indication on the search page that their default filter settings have been applied.

Personally, having all appropriate areas of the site filtered by default would be fantastic as I am only interested in one modloader.

**Comment by dhi-holo** [comment]
Being able to specifically follow a mod with notifications set for only certain variables, such as the modloader and Minecraft versions could be quite helpful in some situations. For example, the mod EMI did an update today as of writing, and uploaded 8 builds. Being able to not get the Forge notifications, halving that, as well as disabling notifications for 1.19.3 (as a version of Minecraft I don't play) would be a nice way to cleanly sort the notifications for only the ones which matter (which for this example, would be 1.19.2, 1.19.4, and 1.20.1 Fabric builds of EMI)

This could maybe be done with a small option next to the following button, or on reclicking it it brings up a input box similar to the ones on the top of the Versions page?
